server/src/test/java/org/elasticsearch/env/NodeRepurposeCommandTests.java

```diff
@@ -0,0 +1,325 @@
+        if (verbose) {
+            terminal.setVerbosity(Terminal.Verbosity.VERBOSE);
+        }
+
+        consumer.accept(terminal);
+
+        assertThat(terminal.getOutput(), outputMatcher);
+
+        expectThrows(IllegalStateException.class, "Must consume input", () -> terminal.readText(""));
+    }
+
+    private static void executeRepurposeCommand(MockTerminal terminal, Settings settings, int ordinal) throws Exception {
+        NodeRepurposeCommand nodeRepurposeCommand = new NodeRepurposeCommand();
+        OptionSet options = nodeRepurposeCommand.getParser()
+            .parse(ordinal != 0 ? new String[]{"--ordinal", Integer.toString(ordinal)} : new String[0]);
+        Environment env = TestEnvironment.newEnvironment(settings);
+        nodeRepurposeCommand.testExecute(terminal, options, env);
+    }
+
+    private Manifest createManifest(Index index) throws org.elasticsearch.gateway.WriteStateException {
+        Manifest manifest = new Manifest(randomIntBetween(1,100), randomIntBetween(1,100), randomIntBetween(1,100),
+            index != null ? Collections.singletonMap(index, randomLongBetween(1,100)) : Collections.emptyMap());
+        Manifest.FORMAT.writeAndCleanup(manifest, nodePaths);
+        return manifest;
+    }
+
+    private Manifest loadManifest() throws IOException {
+        return Manifest.FORMAT.loadLatestState(logger, new NamedXContentRegistry(ClusterModule.getNamedXWriteables()), nodePaths);
+    }
+
+    private void assertManifestIdenticalExceptIndices(Manifest oldManifest, Manifest newManifest) {
+        assertEquals(oldManifest.getGlobalGeneration(), newManifest.getGlobalGeneration());
+        assertEquals(oldManifest.getClusterStateVersion(), newManifest.getClusterStateVersion());
+        assertEquals(oldManifest.getCurrentTerm(), newManifest.getCurrentTerm());
+    }
+
+    private void createIndexDataFiles(Settings settings, boolean includeShardData) throws IOException {
+        String shardDataDirName = Integer.toString(randomInt(10));
+
+        try (NodeEnvironment env = new NodeEnvironment(settings, TestEnvironment.newEnvironment(settings))) {
+            IndexMetaData.FORMAT.write(IndexMetaData.builder(INDEX.getName())
+                .settings(Settings.builder().put("index.version.created", Version.CURRENT))
+                .numberOfShards(1)
+                .numberOfReplicas(1)
+                .build(), env.indexPaths(INDEX));
+            for (Path path : env.indexPaths(INDEX)) {
+                if (includeShardData) {
+                    Files.createDirectories(path.resolve(shardDataDirName));
+                }
+            }
+        }
+    }
+
+    private void verifyUnchangedDataFiles(CheckedRunnable<? extends Exception> runnable) throws Exception {
+        long before = digestPaths();
+        runnable.run();
+        long after = digestPaths();
+        assertEquals("Must not touch files", before, after);
+    }
+
+    private long digestPaths() {
```

Comment:
I wonder if we already have a similar function implemented somewhere (it takes 40 lines of code here) or probably it makes sense to add it to some util class. Probably @ywelsch can suggest something.

Reply:
I found none, will await feedback from @ywelsch on this.

Reply:
I'm not aware of any such existing functionality. There is `BufferedChecksumIndexInput` in Lucene, but that will not shorten the code here.